<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Accessing the Human Cell Atlas datasets • HCAData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Accessing the Human Cell Atlas datasets">
<meta property="og:description" content="HCAData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HCAData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/hcadata.html">Accessing the Human Cell Atlas datasets</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/federicomarini/HCAData/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="hcadata_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Accessing the Human Cell Atlas datasets</h1>
                        <h4 class="author">Federico Marini</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz<br><a class="author_email" href="mailto:#"></a><a href="mailto:marinif@uni-mainz.de" class="email">marinif@uni-mainz.de</a>
      </address>
                  
            <h4 class="date">3 February 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/federicomarini/HCAData/blob/master/vignettes/hcadata.Rmd"><code>vignettes/hcadata.Rmd</code></a></small>
      <div class="hidden name"><code>hcadata.Rmd</code></div>

    </div>

    
    
<p><strong>Compiled date</strong>: 2021-02-03</p>
<p><strong>Last edited</strong>: 2019-05-24</p>
<p><strong>License</strong>: MIT + file LICENSE</p>
<div id="introducing-the-hcadata-package" class="section level1">
<h1 class="hasAnchor">
<a href="#introducing-the-hcadata-package" class="anchor"></a>Introducing the <code>HCAData</code> package</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.13/HCAData">HCAData</a></em> package allows a direct access to the dataset generated by the Human Cell Atlas project for further processing in R and Bioconductor. It does so by providing the datasets as <em><a href="https://bioconductor.org/packages/3.13/SingleCellExperiment">SingleCellExperiment</a></em> objects, i.e. a format which is both efficient and very widely adopted throughout many existing Bioconductor workflows.</p>
<p>The datasets are otherwise available in other formats (also as raw data) at this link: <a href="http://preview.data.humancellatlas.org/" class="uri">http://preview.data.humancellatlas.org/</a>.</p>
</div>
<div id="installing-hcadata" class="section level1">
<h1 class="hasAnchor">
<a href="#installing-hcadata" class="anchor"></a>Installing <code>HCAData</code>
</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.13/HCAData">HCAData</a></em> package can be installed in the conventional way via <em><a href="https://CRAN.R-project.org/package=BiocManager">BiocManager</a></em>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"HCAData"</span><span class="op">)</span></code></pre></div>
<p>This package makes extensive use of the <em><a href="https://bioconductor.org/packages/3.13/HDF5Array">HDF5Array</a></em> package to avoid loading the entire data set in memory. Instead, it stores the counts on disk as a HDF5 file, and loads subsets of the data into memory upon request.</p>
</div>
<div id="loading-hcadata-and-the-human-cell-atlas-data" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-hcadata-and-the-human-cell-atlas-data" class="anchor"></a>Loading <code>HCAData</code> and the Human Cell Atlas data</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/federicomarini/HCAData">"HCAData"</a></span><span class="op">)</span></code></pre></div>
<p>We use the <code>HCAData</code> function to download the relevant files from Bioconductor’s <em><a href="https://bioconductor.org/packages/3.13/ExperimentHub">ExperimentHub</a></em> web resource. If no argument is provided, a list of the available datasets is returned, specifying which name to enter as <code>dataset</code> parameter when calling <code>HCAData</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/HCAData.html">HCAData</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>The list of relevant files includes the HDF5 file containing the counts, as well as the metadata on the rows (genes) and columns (cells).</p>
<p>The output is a single <code>SingleCellExperiment</code> object from the <em><a href="https://bioconductor.org/packages/3.13/SingleCellExperiment">SingleCellExperiment</a></em> package.</p>
<p>Being based on <em><a href="https://bioconductor.org/packages/3.13/ExperimentHub">ExperimentHub</a></em>, the data related to this package can be accessed and queried directly using the package name. Retrieval is then as easy as using their ExperimentHub accession numbers (for the single components of each set), or by using the convenience function provided in this package.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"ExperimentHub"</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"SingleCellExperiment"</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu">query</span><span class="op">(</span><span class="va">eh</span>, <span class="st">"HCAData"</span><span class="op">)</span>
<span class="co">#&gt; ExperimentHub with 6 records</span>
<span class="co">#&gt; # snapshotDate(): 2021-01-28</span>
<span class="co">#&gt; # $dataprovider: Human Cell Atlas</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: character</span>
<span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span>
<span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span>
<span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span>
<span class="co">#&gt; # retrieve records with, e.g., 'object[["EH2047"]]' </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;            title                                                              </span>
<span class="co">#&gt;   EH2047 | Human Cell Atlas - Census of Immune Cells, Bone marrow, 'dense m...</span>
<span class="co">#&gt;   EH2048 | Human Cell Atlas - Census of Immune Cells, Bone marrow, sample (...</span>
<span class="co">#&gt;   EH2049 | Human Cell Atlas - Census of Immune Cells, Bone marrow, gene (ro...</span>
<span class="co">#&gt;   EH2050 | Human Cell Atlas - Census of Immune Cells, Umbilical cord blood,...</span>
<span class="co">#&gt;   EH2051 | Human Cell Atlas - Census of Immune Cells, Umbilical cord blood,...</span>
<span class="co">#&gt;   EH2052 | Human Cell Atlas - Census of Immune Cells, Umbilical cord blood,...</span>

<span class="co"># these three are the components to the bone marrow dataset</span>
<span class="va">bonemarrow_h5densematrix</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH2047"</span><span class="op">]</span><span class="op">]</span>
<span class="va">bonemarrow_coldata</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH2048"</span><span class="op">]</span><span class="op">]</span>
<span class="va">bonemarrow_rowdata</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH2049"</span><span class="op">]</span><span class="op">]</span>

<span class="co"># and are put together when calling...</span>
<span class="va">sce_bonemarrow</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HCAData.html">HCAData</a></span><span class="op">(</span><span class="st">"ica_bone_marrow"</span><span class="op">)</span>
<span class="va">sce_bonemarrow</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 33694 378000 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(33694): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span>
<span class="co">#&gt;   ENSG00000268674</span>
<span class="co">#&gt; rowData names(2): ID Symbol</span>
<span class="co">#&gt; colnames(378000): MantonBM1_HiSeq_1-AAACCTGAGCAGGTCA-1</span>
<span class="co">#&gt;   MantonBM1_HiSeq_1-AAACCTGCACACTGCG-1 ...</span>
<span class="co">#&gt;   MantonBM8_HiSeq_8-TTTGTCATCTGCCAGG-1</span>
<span class="co">#&gt;   MantonBM8_HiSeq_8-TTTGTCATCTTGAGAC-1</span>
<span class="co">#&gt; colData names(1): Barcode</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span>

<span class="co"># similarly, to access the umbilical cord blood dataset</span>
<span class="va">sce_cordblood</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HCAData.html">HCAData</a></span><span class="op">(</span><span class="st">"ica_cord_blood"</span><span class="op">)</span>
<span class="va">sce_cordblood</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 33694 384000 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(33694): ENSG00000243485 ENSG00000237613 ... ENSG00000277475</span>
<span class="co">#&gt;   ENSG00000268674</span>
<span class="co">#&gt; rowData names(2): ID Symbol</span>
<span class="co">#&gt; colnames(384000): MantonCB1_HiSeq_1-AAACCTGAGGAGTTGC-1</span>
<span class="co">#&gt;   MantonCB1_HiSeq_1-AAACCTGAGGCATTGG-1 ...</span>
<span class="co">#&gt;   MantonCB8_HiSeq_8-TTTGTCATCCAGATCA-1</span>
<span class="co">#&gt;   MantonCB8_HiSeq_8-TTTGTCATCGGTCTAA-1</span>
<span class="co">#&gt; colData names(1): Barcode</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(0):</span></code></pre></div>
</div>
<div id="explore-data-with-isee" class="section level1">
<h1 class="hasAnchor">
<a href="#explore-data-with-isee" class="anchor"></a>Explore data with <code>iSEE</code>
</h1>
<p>The datasets are provided in the form of a <code>SingleCellExperiment</code> object. A natural companion to this data structure is the <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> package, which can be used for interactive and reproducible data exploration.</p>
<p>Any analysis steps should be performed in advance before calling <code>iSEE</code>, and since these datasets can be quite big, the operations can be time consuming, and/or require a considerable amount of resources.</p>
<div id="processing-a-subset-of-the-hca-bone-marrow-data" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-a-subset-of-the-hca-bone-marrow-data" class="anchor"></a>Processing a subset of the HCA bone marrow data</h2>
<p>For the scope of the vignette, we subset some cells in the bone marrow dataset to reduce the runtime, and apply some of the steps one would ideally follow in analysing droplet based datasets. For more information on how to properly process such datasets, please refer to the amazing set of resources available in the <em><a href="https://bioconductor.org/packages/3.13/simpleSingleCell">simpleSingleCell</a></em> workflow package.</p>
<p>In brief: we start loading the required libraries for preprocessing, and taking a subset of the bone marrow dataset</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"scran"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LTLA/BiocSingular">"BiocSingular"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://bioconductor.org/packages/scater/">"scater"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"scuttle"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce_bonemarrow</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">sce_bonemarrow</span><span class="op">)</span><span class="op">)</span>, <span class="fl">1000</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<p>First, we relabel the rows with the gene symbols for easier reading with the <code><a href="https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html">uniquifyFeatureNames()</a></code> function from <em><a href="https://bioconductor.org/packages/3.13/scater">scater</a></em>. Then we compute some QC metrics and add these to the original <code>sce</code> object, using <code><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQC.html">addPerCellQC()</a></code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html">uniquifyFeatureNames</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">ID</span>, <span class="fu">rowData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">Symbol</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "RP11-34P13.3"  "FAM138A"       "OR4F5"         "RP11-34P13.7" </span>
<span class="co">#&gt; [5] "RP11-34P13.8"  "RP11-34P13.14"</span>

<span class="va">is.mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"MT-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scuttle</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/addPerCellQC.html">addPerCellQC</a></span><span class="op">(</span><span class="va">sce</span>, subsets<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Mito<span class="op">=</span><span class="va">is.mito</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>We proceed with normalization, performed with the deconvolution method implemented in <em><a href="https://bioconductor.org/packages/3.13/scran">scran</a></em> - a pre-clustering step is done in advance, to avoid pooling cells that are very different between each other.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lib.sf.bonemarrow</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/librarySizeFactors.html">librarySizeFactors</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">lib.sf.bonemarrow</span><span class="op">)</span>
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;  0.0157  0.3407  0.6291  1.0000  0.9087 18.0304</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span>
<span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/quickCluster.html">quickCluster</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span>
<span class="co">#&gt; clusters</span>
<span class="co">#&gt;   1   2   3   4   5 </span>
<span class="co">#&gt; 137 232 207 239 185</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/computeSumFactors.html">computeSumFactors</a></span><span class="op">(</span><span class="va">sce</span>, min.mean<span class="op">=</span><span class="fl">0.1</span>, cluster<span class="op">=</span><span class="va">clusters</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html">logNormCounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>
<span class="fu">assayNames</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>
<span class="co">#&gt; [1] "counts"    "logcounts"</span></code></pre></div>
<p>In the following lines of code, we model the mean-variance trend (with technical noise as Poisson), to extract the biological component of the variance for the genes under inspection.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dec.bonemarrow</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVarByPoisson.html">modelGeneVarByPoisson</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>
<span class="va">top.dec</span> <span class="op">&lt;-</span> <span class="va">dec.bonemarrow</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">dec.bonemarrow</span><span class="op">$</span><span class="va">bio</span>, decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span> 
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">top.dec</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 6 rows and 6 columns</span>
<span class="co">#&gt;             mean     total      tech       bio   p.value       FDR</span>
<span class="co">#&gt;        &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; S100A8   1.09003   5.19285   1.03231   4.16054         0         0</span>
<span class="co">#&gt; S100A9   1.08007   4.96772   1.02675   3.94097         0         0</span>
<span class="co">#&gt; MALAT1   7.23554   4.50786   0.69933   3.80853         0         0</span>
<span class="co">#&gt; LYZ      1.12696   4.79231   1.05247   3.73984         0         0</span>
<span class="co">#&gt; HBB      1.07098   4.35318   1.02161   3.33157         0         0</span>
<span class="co">#&gt; RPS27    4.96094   3.76764   1.22724   2.54039         0         0</span>

<span class="va">fit.bonemarrow</span> <span class="op">&lt;-</span> <span class="fu">metadata</span><span class="op">(</span><span class="va">dec.bonemarrow</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">fit.bonemarrow</span><span class="op">$</span><span class="va">mean</span>, <span class="va">fit.bonemarrow</span><span class="op">$</span><span class="va">var</span>, xlab<span class="op">=</span><span class="st">"Mean of log-expression"</span>,
    ylab<span class="op">=</span><span class="st">"Variance of log-expression"</span>, pch <span class="op">=</span> <span class="fl">16</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/curve.html">curve</a></span><span class="op">(</span><span class="va">fit.bonemarrow</span><span class="op">$</span><span class="fu">trend</span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, col<span class="op">=</span><span class="st">"dodgerblue"</span>, add<span class="op">=</span><span class="cn">TRUE</span>, lwd<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="hcadata_files/figure-html/hvg-1.png" width="700"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotExpression.html">plotExpression</a></span><span class="op">(</span><span class="va">sce</span>, features<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">top.dec</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="hcadata_files/figure-html/hvg-2.png" width="700"></p>
<p>With the <code>denoisePCA</code> function from <em><a href="https://bioconductor.org/packages/3.13/scran">scran</a></em>, we can compute a reduced dimension view of the data, accounting for the Poisson technical trend.</p>
<p>Once we computed the PCA, we provide that as initialization to <code>runTSNE</code>, and obtain the t-SNE results, stored in the appropriate slot of our <code>sce</code> object.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hvg.bonemarrow</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html">getTopHVGs</a></span><span class="op">(</span><span class="va">dec.bonemarrow</span>, prop <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/denoisePCA.html">denoisePCA</a></span><span class="op">(</span><span class="va">sce</span>, 
                  technical<span class="op">=</span><span class="va">dec.bonemarrow</span>, 
                  subset.row <span class="op">=</span> <span class="va">hvg.bonemarrow</span>, 
                  BSPARAM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/BiocSingularParam.html">IrlbaParam</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDim</a></span><span class="op">(</span><span class="va">sce</span>, <span class="st">"PCA"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 5</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/attr.html">attr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDim</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>, <span class="st">"percentVar"</span><span class="op">)</span>, xlab<span class="op">=</span><span class="st">"PC"</span>,
     ylab<span class="op">=</span><span class="st">"Proportion of variance explained"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>v<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDim</a></span><span class="op">(</span><span class="va">sce</span>, <span class="st">"PCA"</span><span class="op">)</span><span class="op">)</span>, lty<span class="op">=</span><span class="fl">2</span>, col<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></code></pre></div>
<p><img src="hcadata_files/figure-html/dimred-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html">plotPCA</a></span><span class="op">(</span><span class="va">sce</span>, ncomponents<span class="op">=</span><span class="fl">3</span>, colour_by<span class="op">=</span><span class="st">"subsets_Mito_percent"</span><span class="op">)</span></code></pre></div>
<p><img src="hcadata_files/figure-html/dimred-2.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runTSNE.html">runTSNE</a></span><span class="op">(</span><span class="va">sce</span>, dimred<span class="op">=</span><span class="st">"PCA"</span>, perplexity<span class="op">=</span><span class="fl">30</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html">plotTSNE</a></span><span class="op">(</span><span class="va">sce</span>, colour_by<span class="op">=</span><span class="st">"subsets_Mito_percent"</span><span class="op">)</span></code></pre></div>
<p><img src="hcadata_files/figure-html/dimred-3.png" width="700"></p>
<p>After this, we can compute a shared nearest neighbour graph to identify clusters in our dataset. Once the cluster memberships are defined, we assign this vector to a corresponding <code>colData</code> slot, and plot the t-SNE for our subset, coloured accordingly.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snn.gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html">buildSNNGraph</a></span><span class="op">(</span><span class="va">sce</span>, use.dimred<span class="op">=</span><span class="st">"PCA"</span><span class="op">)</span>
<span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_walktrap.html">cluster_walktrap</a></span><span class="op">(</span><span class="va">snn.gr</span><span class="op">)</span>
<span class="va">sce</span><span class="op">$</span><span class="va">Cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">$</span><span class="va">membership</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">Cluster</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   1   2   3   4   5   6   7   8   9  10  11  12  13 </span>
<span class="co">#&gt;  62 218 109  96 121  56  41  62  55  82  30  34  34</span>
<span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html">plotTSNE</a></span><span class="op">(</span><span class="va">sce</span>, colour_by<span class="op">=</span><span class="st">"Cluster"</span><span class="op">)</span></code></pre></div>
<p><img src="hcadata_files/figure-html/clusters-1.png" width="700"></p>
<p>Even if we only took a subset of the available full data, we can observe that different clusters are indeed nicely separated. Subsequent steps would then involve identification of marker genes, and more advanced downstream techniques, which are not part of the scope of this vignette. To read more on what can be done, the vignettes of the <em><a href="https://bioconductor.org/packages/3.13/simpleSingleCell">simpleSingleCell</a></em> workflow package are an excellent place to start.</p>
</div>
<div id="exploring-the-dataset-with-isee" class="section level2">
<h2 class="hasAnchor">
<a href="#exploring-the-dataset-with-isee" class="anchor"></a>Exploring the dataset with <code>iSEE</code>
</h2>
<p>Once the processing steps above are done, we can call <code>iSEE</code> with the subsampled <code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEE">iSEE</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html">iSEE</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div id="saving-the-processed-object" class="section level2">
<h2 class="hasAnchor">
<a href="#saving-the-processed-object" class="anchor"></a>Saving the processed object</h2>
<p>You can save the <code>sce</code> object to a serialized R object with</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">destination</span> <span class="op">&lt;-</span> <span class="st">"where/to/store/the/processed/data.rds"</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">sce</span>, file <span class="op">=</span> <span class="va">destination</span><span class="op">)</span></code></pre></div>
<p>The object can be read into a new R session with <code><a href="https://rdrr.io/r/base/readRDS.html">readRDS(destination)</a></code>, provided the HDF5 file remains in its original location (conveniently stored in the default location of <code>ExperimentHub</code>).</p>
</div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R Under development (unstable) (2021-02-02 r79929)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] scuttle_1.1.9               scater_1.19.9              </span>
<span class="co">#&gt;  [3] ggplot2_3.3.3               BiocSingular_1.7.2         </span>
<span class="co">#&gt;  [5] scran_1.19.10               rhdf5_2.35.0               </span>
<span class="co">#&gt;  [7] ExperimentHub_1.17.0        AnnotationHub_2.23.1       </span>
<span class="co">#&gt;  [9] BiocFileCache_1.15.1        dbplyr_2.0.0               </span>
<span class="co">#&gt; [11] HCAData_1.7.1               SingleCellExperiment_1.13.6</span>
<span class="co">#&gt; [13] SummarizedExperiment_1.21.1 Biobase_2.51.0             </span>
<span class="co">#&gt; [15] GenomicRanges_1.43.3        GenomeInfoDb_1.27.5        </span>
<span class="co">#&gt; [17] IRanges_2.25.6              S4Vectors_0.29.6           </span>
<span class="co">#&gt; [19] BiocGenerics_0.37.0         MatrixGenerics_1.3.1       </span>
<span class="co">#&gt; [21] matrixStats_0.58.0          BiocStyle_2.19.1           </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] Rtsne_0.15                    ggbeeswarm_0.6.0             </span>
<span class="co">#&gt;   [3] colorspace_2.0-0              ellipsis_0.3.1               </span>
<span class="co">#&gt;   [5] rprojroot_2.0.2               bluster_1.1.5                </span>
<span class="co">#&gt;   [7] XVector_0.31.1                BiocNeighbors_1.9.4          </span>
<span class="co">#&gt;   [9] fs_1.5.0                      farver_2.0.3                 </span>
<span class="co">#&gt;  [11] bit64_4.0.5                   interactiveDisplayBase_1.29.0</span>
<span class="co">#&gt;  [13] AnnotationDbi_1.53.0          sparseMatrixStats_1.3.2      </span>
<span class="co">#&gt;  [15] cachem_1.0.1                  knitr_1.31                   </span>
<span class="co">#&gt;  [17] shiny_1.6.0                   HDF5Array_1.19.0             </span>
<span class="co">#&gt;  [19] BiocManager_1.30.10           compiler_4.1.0               </span>
<span class="co">#&gt;  [21] httr_1.4.2                    dqrng_0.2.1                  </span>
<span class="co">#&gt;  [23] assertthat_0.2.1              Matrix_1.3-2                 </span>
<span class="co">#&gt;  [25] fastmap_1.1.0                 limma_3.47.6                 </span>
<span class="co">#&gt;  [27] later_1.1.0.1                 htmltools_0.5.1.1            </span>
<span class="co">#&gt;  [29] tools_4.1.0                   rsvd_1.0.3                   </span>
<span class="co">#&gt;  [31] igraph_1.2.6                  gtable_0.3.0                 </span>
<span class="co">#&gt;  [33] glue_1.4.2                    GenomeInfoDbData_1.2.4       </span>
<span class="co">#&gt;  [35] dplyr_1.0.4                   rappdirs_0.3.3               </span>
<span class="co">#&gt;  [37] Rcpp_1.0.6                    pkgdown_1.6.1.9000           </span>
<span class="co">#&gt;  [39] vctrs_0.3.6                   rhdf5filters_1.3.3           </span>
<span class="co">#&gt;  [41] DelayedMatrixStats_1.13.4     xfun_0.20                    </span>
<span class="co">#&gt;  [43] stringr_1.4.0                 beachmat_2.7.6               </span>
<span class="co">#&gt;  [45] mime_0.9                      lifecycle_0.2.0              </span>
<span class="co">#&gt;  [47] irlba_2.3.3                   statmod_1.4.35               </span>
<span class="co">#&gt;  [49] edgeR_3.33.1                  zlibbioc_1.37.0              </span>
<span class="co">#&gt;  [51] scales_1.1.1                  ragg_0.4.1                   </span>
<span class="co">#&gt;  [53] promises_1.1.1                yaml_2.2.1                   </span>
<span class="co">#&gt;  [55] curl_4.3                      gridExtra_2.3                </span>
<span class="co">#&gt;  [57] memoise_2.0.0                 stringi_1.5.3                </span>
<span class="co">#&gt;  [59] RSQLite_2.2.3                 highr_0.8                    </span>
<span class="co">#&gt;  [61] BiocVersion_3.13.1            desc_1.2.0                   </span>
<span class="co">#&gt;  [63] ScaledMatrix_0.99.2           filelock_1.0.2               </span>
<span class="co">#&gt;  [65] BiocParallel_1.25.3           rlang_0.4.10                 </span>
<span class="co">#&gt;  [67] pkgconfig_2.0.3               systemfonts_1.0.0            </span>
<span class="co">#&gt;  [69] bitops_1.0-6                  evaluate_0.14                </span>
<span class="co">#&gt;  [71] lattice_0.20-41               purrr_0.3.4                  </span>
<span class="co">#&gt;  [73] Rhdf5lib_1.13.0               labeling_0.4.2               </span>
<span class="co">#&gt;  [75] bit_4.0.4                     tidyselect_1.1.0             </span>
<span class="co">#&gt;  [77] magrittr_2.0.1                bookdown_0.21                </span>
<span class="co">#&gt;  [79] R6_2.5.0                      generics_0.1.0               </span>
<span class="co">#&gt;  [81] metapod_0.99.5                DelayedArray_0.17.7          </span>
<span class="co">#&gt;  [83] DBI_1.1.1                     pillar_1.4.7                 </span>
<span class="co">#&gt;  [85] withr_2.4.1                   RCurl_1.98-1.2               </span>
<span class="co">#&gt;  [87] tibble_3.0.6                  crayon_1.4.0                 </span>
<span class="co">#&gt;  [89] rmarkdown_2.6                 viridis_0.5.1                </span>
<span class="co">#&gt;  [91] locfit_1.5-9.4                grid_4.1.0                   </span>
<span class="co">#&gt;  [93] blob_1.2.1                    digest_0.6.27                </span>
<span class="co">#&gt;  [95] xtable_1.8-4                  httpuv_1.5.5                 </span>
<span class="co">#&gt;  [97] textshaping_0.2.1             munsell_0.5.0                </span>
<span class="co">#&gt;  [99] viridisLite_0.3.0             beeswarm_0.2.3               </span>
<span class="co">#&gt; [101] vipor_0.4.5</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Federico Marini.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
